planner/cascades/transformation_rules.go

```diff
@@ -1781,3 +1783,123 @@ func isInnerJoinKeysContainUniqueKey(innerGroup *memo.Group, joinKeys *expressio
 	}
 	return false, nil
 }
+
+// TransformAggregateCaseToSelection convert Agg(case when) to Agg->Selection.
+type TransformAggregateCaseToSelection struct {
+	baseRule
+}
+
+// NewRuleTransformAggregateCaseToSelection creates a new Transformation TransformAggregateCaseToSelection.
+// The pattern of this rule is `Agg->X`.
+func NewRuleTransformAggregateCaseToSelection() Transformation {
+	rule := &TransformAggregateCaseToSelection{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandAggregation,
+		memo.EngineTiDBOnly,
+	)
+	return rule
+}
+
+// Match implements Transformation interface.
+func (r *TransformAggregateCaseToSelection) Match(expr *memo.ExprIter) bool {
+	agg := expr.GetExpr().ExprNode.(*plannercore.LogicalAggregation)
+	return len(agg.GetGroupByUsedCols()) == 0 && len(agg.AggFuncs) == 1 && len(agg.AggFuncs[0].Args) == 1 && r.isTwoOrThreeArgCase(agg.AggFuncs[0].Args[0])
```

Comment:
I think use `len(agg.GroupByItems) == 0` is ok? Do we have other cases?

Reply:
group by constant? 
such as  select xxx from t group by 'a'

Reply:
In `select xxx from t group by concat(a,c)`, `len(agg.GroupByItems) == 0` is true.

Reply:
But if we group by a constant, we cannot use this optimization?

Reply:
`group by constant, no group by, group by func(constant...)` all is ok

Reply:
See the example:
```
mysql> select * from t;
+------+------+
| a    | b    |
+------+------+
|    1 |    1 |
+------+------+
1 row in set (0.00 sec)

mysql> select sum(b) from t where a > 1 group by 'p';
Empty set (0.00 sec)

mysql> select sum(case when a > 1 then b else 0 end) from t group by 'p';
+----------------------------------------+
| sum(case when a > 1 then b else 0 end) |
+----------------------------------------+
|                                      0 |
+----------------------------------------+
1 row in set (0.00 sec)
```

Reply:
ok....

Reply:
yes, `len(agg.GroupByItems) == 0` is right